src/main/java/vazkii/psi/client/core/handler/ShaderHandler.java

```diff
@@ -161,56 +160,12 @@ private static String getLogInfo(int obj) {
 	}
 
 	private static String readFileAsString(String filename) throws Exception {
-		StringBuilder source = new StringBuilder();
-		InputStream in = ShaderHandler.class.getResourceAsStream(filename);
-		Exception exception = null;
-		BufferedReader reader;
-
-		if (in == null) {
-			return "";
-		}
-
-		try {
-			reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
-
-			Exception innerExc = null;
-			try {
-				String line;
-				while ((line = reader.readLine()) != null) {
-					source.append(line).append('\n');
-				}
-			} catch (Exception exc) {
-				exception = exc;
-			} finally {
-				try {
-					reader.close();
-				} catch (Exception exc) {
-					innerExc = exc;
-				}
-			}
-
-			if (innerExc != null) {
-				throw innerExc;
-			}
-		} catch (Exception exc) {
-			exception = exc;
-		} finally {
-			try {
-				in.close();
-			} catch (Exception exc) {
-				if (exception == null) {
-					exception = exc;
-				} else {
-					exc.printStackTrace();
-				}
+		try (InputStream in = ShaderHandler.class.getResourceAsStream(filename)) {
```

Comment:
using a try-with-resources here was also a PMD suggestion, nooooo idea what the hell was going on before